staging/src/k8s.io/pod-security-admission/admission/admission.go

```diff
@@ -190,36 +199,39 @@ func (a *Admission) ValidateConfiguration() error {
 
 // Validate admits an API request.
 // The objects in admission attributes are expected to be external v1 objects that we care about.
-func (a *Admission) Validate(ctx context.Context, attrs Attributes) *admissionv1.AdmissionResponse {
-	var response *admissionv1.AdmissionResponse
+func (a *Admission) Validate(ctx context.Context, attrs admissionapi.Attributes) *admissionv1.AdmissionResponse {
+	var response *admissionapi.Response
+	var observe = true
 	switch attrs.GetResource().GroupResource() {
 	case corev1.Resource("namespaces"):
+		observe = false
 		response = a.ValidateNamespace(ctx, attrs)
 	case corev1.Resource("pods"):
 		response = a.ValidatePod(ctx, attrs)
 	default:
 		response = a.ValidatePodController(ctx, attrs)
 	}
-
-	// TODO: record metrics.
-
-	return response
+	if a.Metrics != nil && observe {
+		a.Metrics.RecordEvaluation(attrs, response)
+	}
```

Comment:
I'm afraid that recording the metrics here might not be viable, at least not without plumbing through a lot of additional complexity. I think we might just need to record the evaluation at every evaluation & return site.

That also means we no longer need the `admissionapi.Response` type.

Reply:
I think we can use a defer function in Validate{Pod|Namespace|PodController} just like
```
defer func() {
		if a.Metrics != nil{
			if a.defaultPolicy.Audit.Valid(){
				a.Metrics.RecordEvaluation(metrics.DecisionAllow, a.defaultPolicy.Audit, metrics.ModeAudit, attrs)
			}
			if a.defaultPolicy.Warn.Valid() {
				a.Metrics.RecordEvaluation(metrics.DecisionAllow, a.defaultPolicy.Warn, metrics.ModeWarn, attrs)
			}
			if a.defaultPolicy.Enforce.Valid(){
				a.Metrics.RecordEvaluation(metrics.DecisionAllow, a.defaultPolicy.Enforce, metrics.ModeEnforce, attrs)
			}
		}
	}()
```
What do you think? :) @tallclair